### PR TITLE
[TASK-7278] feat: remember selected address

### PR DESCRIPTION
**Pull request by jjramirezn**
## Summary by CodeRabbit

- **New Features**
	- Enhanced wallet address handling for improved user experience.
	- Selected wallet address now persists across sessions using user preferences.
	- Utility functions for local storage operations are now publicly accessible.
	- Improved initialization of token data based on user preferences.
	- Enhanced error handling for user preferences updates.

- **Bug Fixes**
	- Error logging added for balance fetching issues to assist in troubleshooting.

### CODE REVIEW

File: src/context/walletContext/walletContext.tsx [hunk 1]
```diff
@@ -8,7 +8,13 @@ import { useQuery, useQueryClient } from '@tanstack/react-query'
 import { PEANUT_WALLET_CHAIN, PEANUT_WALLET_TOKEN } from '@/constants'
 import { Chain, erc20Abi, getAddress, parseUnits } from 'viem'
 import { useAuth } from '../authContext'
-import { backgroundColorFromAddress, areEvmAddressesEqual, fetchWalletBalances } from '@/utils'
+import {
+    backgroundColorFromAddress,
+    areEvmAddressesEqual,
+    fetchWalletBalances,
+    getFromLocalStorage,
+    saveToLocalStorage,
```
Comment (Hugo0) [inline src/context/walletContext/walletContext.tsx]: good on using existing functions!

Thought: we should clean up general.utils.ts, we save tons of shit to localStorage in random ways. Need to clean this up - probably same time we redo our tx indexing

**Review by Hugo0 — CHANGES_REQUESTED** [review]
Issue, blocking: in this PR we set a specific key "selectedAddress" to localStorage. I think this can lead to localStorage key pollution, and it's not clear what the item means, on its own.

Since last selected address wont be the only thing we should store for users for convenience, e.g. they might have a preffered token or chain, or they might want dark or light mode, its preferable to group these all in one. We _do_ have a `getPeanutPreferences` & `updatePeanutPreferences`. Right now it's pretty empty  and the 3 items it has are badly named, but imo we should make use of a user-preferences type of pattern. Wdyt?

File: src/context/tokenSelector.context.tsx [hunk 2]
```diff
@@ -59,22 +59,17 @@ export const TokenContextProvider = ({ children }: { children: React.ReactNode }
...
-        setSelectedTokenAddress(
-            preferences?.tokenAddress ? preferences.tokenAddress : '0x0b2C639c533813f4Aa9D7837CAf62653d097Ff85'
-        )
-        setSelectedTokenPrice(undefined)
-        setSelectedTokenDecimals(undefined)
+        const { lastUsedToken } = utils.getUserPreferences() ?? {}
+        const tokenData = lastUsedToken ?? initialTokenData
+        setSelectedChainID(tokenData.chainId)
+        setSelectedTokenAddress(tokenData.address)
+        setSelectedTokenDecimals(tokenData.decimals)
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/context/tokenSelector.context.tsx:72]: done

**Review by Hugo0 — APPROVED** [review]
I think this is great!

**Pull request merged**